### DISCUSSION TITLE
React `use` hook vs `await` for Server Components

### DISCUSSION THREAD

**Discussion by jefer94** [vercel/next.js#73401] [Help] [answered]
### Summary

Does Next provide any improvement for server components if we use `use` hook? I'm awake that `use` offer an alternative to provide promises for Client Components

**Comment by icyJoseph** [comment, marked as answer]
The improvement/benefit comes straight up from React itself. 

- https://react.dev/reference/react/use#streaming-data-from-server-to-client

That one is, IMHO, the biggest win of the `use` hook. That you can establish a ~promise~stream, from Server to client.

```jsx
import { fetchMessage } from './lib.js';
import { Message } from './message.js';

export default function App() {
  const messagePromise = fetchMessage();
  return (
  <>
    <SomeOtherComponent />
    <Suspense fallback={<p>waiting for message...</p>}>
      <Message messagePromise={messagePromise} />
    </Suspense>
  </>
  );
}
```

And 

```jsx
// message.js
'use client';

import { use } from 'react';

export function Message({ messagePromise }) {
  const messageContent = use(messagePromise);
  return <p>Here is the message: {messageContent}</p>;
}
```

That opens the possibility to send HTML that doesn't depend on that promise, upfront, while still being able to show a suspense boundary and what not.

**Reply by jefer94** [reply]
So considering the SSR capabilities of Next, it could be used to reduce the first javascript load like React `lazy` and Next `dynamic` do, showing a skeleton during the process
